lib/CL/devices/pthread/pthread.c

```diff
@@ -117,6 +113,10 @@ struct data {
   mem_regions_management* mem_regions;
 #endif
 
+  /* Lock for command list related operations */
+  pthread_mutex_t cq_lock __attribute__ ((aligned (CACHELINE_SIZE)));
```

Comment:
OK, here CACHELINE_SIZE is used. I think that's too general. Sure 64B is a common choice, but it's not a global truth, if you know what I mean :) Better just have it as a specifically named macro PTHREAD_MUTEX_ALIGNMENT or similar if you have measured it speeds up the lock handling.